src/System.Linq.Expressions/tests/ReadOnlyCollectionBuilder/ReadOnlyCollectionBuilder.Tests.Misc.cs

```diff
@@ -0,0 +1,560 @@
+                list.Add(value);
+                Assert.True(list.Contains(value)); //"Should contain item."
+            }
+
+            public void NonGenericIListBasicContains(T[] items)
+            {
+                ReadOnlyCollectionBuilder<T> list = new ReadOnlyCollectionBuilder<T>(items);
+                IList _ilist = list;
+
+                for (int i = 0; i < items.Length; i++)
+                {
+                    Assert.True(_ilist.Contains(items[i])); //"Should contain item."
+                }
+            }
+
+            public void NonGenericIListNonExistingValues(T[] itemsX, T[] itemsY)
+            {
+                ReadOnlyCollectionBuilder<T> list = new ReadOnlyCollectionBuilder<T>(itemsX);
+                IList _ilist = list;
+
+                for (int i = 0; i < itemsY.Length; i++)
+                {
+                    Assert.False(_ilist.Contains(itemsY[i])); //"Should not contain item"
+                }
+            }
+
+            public void NonGenericIListRemovedValues(T[] items)
+            {
+                ReadOnlyCollectionBuilder<T> list = new ReadOnlyCollectionBuilder<T>(items);
+                IList _ilist = list;
+                for (int i = 0; i < items.Length; i++)
+                {
+                    list.Remove(items[i]);
+                    Assert.False(_ilist.Contains(items[i])); //"Should not contain item"
+                }
+            }
+
+            public void NonGenericIListAddRemoveValues(T[] items)
+            {
+                ReadOnlyCollectionBuilder<T> list = new ReadOnlyCollectionBuilder<T>(items);
+                IList _ilist = list;
+                for (int i = 0; i < items.Length; i++)
+                {
+                    list.Add(items[i]);
+                    list.Remove(items[i]);
+                    list.Add(items[i]);
+                    Assert.True(_ilist.Contains(items[i])); //"Should contain item."
+                }
+            }
+
+            public void NonGenericIListMultipleValues(T[] items, int times)
+            {
+                ReadOnlyCollectionBuilder<T> list = new ReadOnlyCollectionBuilder<T>(items);
+                IList _ilist = list;
+
+                for (int i = 0; i < times; i++)
+                {
+                    list.Add(items[items.Length / 2]);
+                }
+
+                for (int i = 0; i < times + 1; i++)
+                {
+                    Assert.True(_ilist.Contains(items[items.Length / 2])); //"Should contain item."
+                    list.Remove(items[items.Length / 2]);
+                }
+                Assert.False(_ilist.Contains(items[items.Length / 2])); //"Should not contain item"
+            }
+
+            public void NonGenericIListContainsNullWhenReference(T[] items, T value)
+            {
+                if ((object)value != null)
+                {
+                    throw new ArgumentException("invalid argument passed to testcase");
```

Comment:
list.Add(null); can not be compiled
list.Add((T)null); can not be compiled also